qiskit_nature/utils/testing.py

```diff
@@ -0,0 +1,28 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Testing utilities."""
+
+import numpy as np
+
+
+def random_antisymmetric_matrix(dim: int) -> np.ndarray:
+    """Return a random antisymmetric matrix.
+
+    Args:
+        dim: The width and height of the matrix.
+
+    Returns:
+        The sampled antisymmetric matrix.
+    """
+    mat = np.random.randn(dim, dim) + 1j * np.random.randn(dim, dim)
```

Comment:
Usually with random aspects, especially when used in testing, we often want a reproducible entity/result. Can we swithc this over, like the rest of Nature, to use the algorithm_globals random, then seeding that, as a common place. gives us that possibility. The syntax is equivalent since its backed by a numpy generator instance, so
```
from qiskit.utils import algorithm_globals

mat = algorithm_globals.random.randn(.... etc 
```

Reply:
Right, the function should take the seed as input. I've done this following the same style as Terra, e.g. https://github.com/Qiskit/qiskit-terra/blob/89c4f6eb5c56f49ec5149e5be39c77d36b182b4e/qiskit/quantum_info/operators/random.py